### PR TITLE
[JENKINS-41467] Include missing capabilities to Folders POs

### PR DESCRIPTION
**Pull request by varyvol**
[JENKINS-41467](https://issues.jenkins-ci.org/browse/JENKINS-41467)

New capabilities included:
- Credentials scoped to a folder.
- Environmental properties.
- View tabs.
- Delete folders.

@reviewbybees

### CODE REVIEW

**Comment by varyvol** [comment]
Tests passing on CI.

**Comment by mafraba** [comment]
🐝

**Comment by varyvol** [comment]
@reviewbybees done.

**Comment by andresrc** [comment]
@olivergondza ok to merge this one?

File: src/main/java/org/jenkinsci/test/acceptance/po/Folder.java [hunk 1]
```diff
@@ -58,4 +69,67 @@ public JobsMixIn getJobs() {
...
+
+    @Override
+    public void delete() {
+        open();
+        clickLink("Delete Folder");
+        waitFor(by.button("Yes")).click();
+    }
+
+    public void setEnvironmentalVariables(final Map<String, String> envVbles) {
+        configure();
```
Comment (olivergondza) [inline src/main/java/org/jenkinsci/test/acceptance/po/Folder.java]: Why does this do configure and save?

File: src/test/java/plugins/FolderPluginTest.java [hunk 2]
```diff
@@ -87,4 +110,116 @@ public void createFolderHierarchy() {
...
+
+        this.checkViews(folder, MY_VIEW);
+
+        myView.delete();
+
+        this.checkViews(folder, ALL_VIEW, ALL_VIEW);
+    }
+
+    /**
+     * Test scenario to validate credentials scoped in folders.
```
Comment (olivergondza) [inline src/test/java/plugins/FolderPluginTest.java]: nit: this is what the method name should say.

**Comment by olivergondza** [comment]
Looks good to me othervise

File: src/main/java/org/jenkinsci/test/acceptance/po/Folder.java (hunk 1 above)
Comment (varyvol) [inline src/main/java/org/jenkinsci/test/acceptance/po/Folder.java]: Not sure what you mean @olivergondza 

Do you mean configure and save should be called from the outside? Or that save should not be called?
I made it that way so you can do folder.setEnvironmentalVariables(xxx) without opening configuration and saving.

File: src/test/java/plugins/FolderPluginTest.java (hunk 2 above)
Comment (varyvol) [inline src/test/java/plugins/FolderPluginTest.java]: Agree. Changed.

File: src/main/java/org/jenkinsci/test/acceptance/po/Folder.java (hunk 1 above)
Comment (olivergondza) [inline src/main/java/org/jenkinsci/test/acceptance/po/Folder.java]: > Do you mean configure and save should be called from the outside?

Yes.

Comment (varyvol) [inline src/main/java/org/jenkinsci/test/acceptance/po/Folder.java]: Removed then.

**Pull request merged**